### PR TITLE
Integrate firebase-sessions with GoogleDataTransport

### PR DESCRIPTION
**Pull request by samedson**
- Adds a dependency on data-transport
 - Adds TestSessionEventData.kt to store Session Event objects used for testing

#no-changelog

### CODE REVIEW

File: firebase-sessions/firebase-sessions.gradle.kts [hunk 1]
```diff
@@ -51,6 +51,8 @@ dependencies {
 
   runtimeOnly("com.google.firebase:firebase-installations:17.1.3")
 
+  implementation("com.google.android.datatransport:transport-api:3.0.0")
+
```
Comment (samedson) [inline firebase-sessions/firebase-sessions.gradle.kts:55]: I also see the following in other SDKs. Do I need them?

```

   implementation("com.google.android.datatransport:transport-runtime:3.1.8")
   implementation("com.google.android.datatransport:transport-backend-cct:3.1.8")
```

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/EventGDTLogger.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      .send(Event.ofData(sessionEvent))
+  }
+
+  private fun encode(value: SessionEvent): ByteArray {
+    return SessionEvents.SESSION_EVENT_ENCODER.encode(value).toByteArray()
+  }
+
+  companion object {
+    // TODO What do we put for the AQS Log Source
+    private const val AQS_LOG_SOURCE = "AQS_LOG_SOURCE"
```
Comment (samedson) [inline firebase-sessions/src/main/kotlin/com/google/firebase/sessions/EventGDTLogger.kt:59]: What do we put for the log source? It seems to be a number on iOS, but a string on Android

File: firebase-sessions/firebase-sessions.gradle.kts (hunk 1 above)
Comment (visumickey) [reply firebase-sessions/firebase-sessions.gradle.kts:55]: I think you don't need `com.google.android.datatransport:transport-backend-cct`. Basically, there is an API component to use the APIs, runtime component to do the actual operation and backend-cct is for clearcut compatibility. We don't need CCT compatibility here.

Here is how Fireperf depends on Firelog. https://github.com/firebase/firebase-android-sdk/blob/master/firebase-perf/firebase-perf.gradle#L104-L105

So either a runtime dependency on their runtime dependency or a pointed dependency to a production version - both are fine.

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/EventGDTLogger.kt (hunk 2 above)
Comment (visumickey) [reply firebase-sessions/src/main/kotlin/com/google/firebase/sessions/EventGDTLogger.kt:59]: I think it needs both LogSourceName and LogSourceNumber. Firelog reference https://github.com/firebase/firebase-android-sdk/blob/master/transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/LogRequest.java#L38-L42.

This is how we did it Fireperf. https://github.com/firebase/firebase-android-sdk/blob/master/firebase-perf/src/main/java/com/google/firebase/perf/config/ConfigurationConstants.java#L581-L590

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionCoordinator.kt [hunk 3]
```diff
@@ -47,6 +48,12 @@ internal class SessionCoordinator(
         }
 
       Log.i(TAG, "Initiate session start: $sessionEvent")
```
Comment (mrober) [inline firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionCoordinator.kt:50]: Can you move this log to after the `.log` in the try below? Then we either get a log of success or failure on the session start being sent.

**Review by mrober — APPROVED** [review]

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionCoordinator.kt (hunk 3 above)
Comment (samedson) [reply firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionCoordinator.kt]: Done

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/EventGDTLogger.kt (hunk 2 above)
Comment (samedson) [reply firebase-sessions/src/main/kotlin/com/google/firebase/sessions/EventGDTLogger.kt]: I see a lot of other SDKs just using the log source name, so I'm trying that out. Gonna send some events in the test app hopefully

File: firebase-sessions/firebase-sessions.gradle.kts [hunk 4]
```diff
@@ -47,9 +47,11 @@ dependencies {
   implementation("com.google.firebase:firebase-encoders-json:18.0.1")
   implementation("com.google.firebase:firebase-encoders:17.0.0")
   implementation("com.google.firebase:firebase-installations-interop:17.1.0")
+  implementation("com.google.android.datatransport:transport-api:3.0.0")
   implementation(libs.androidx.annotation)
 
   runtimeOnly("com.google.firebase:firebase-installations:17.1.3")
+  runtimeOnly(project(":firebase-datatransport"))
```
Comment (samedson) [inline firebase-sessions/firebase-sessions.gradle.kts:54]: This was required to make FireLog work

**Pull request merged**